guides/source/documents.yaml

```diff
@@ -72,6 +72,10 @@
       url: active_support_core_extensions.html
       description: This guide documents the Ruby core extensions defined in Active Support.
     -
+      name: Active Support On Load Hooks
```

Comment:
Maybe instead of adding a new guide we should put it as a topic the initialization.html guide? @fxn WDYT?